src/redux/todosSlice.js

```diff
@@ -0,0 +1,23 @@
+import { createSlice } from "@reduxjs/toolkit";
+
+const todosSlice = createSlice({
+  name: "todos",
+  initialState: [],
+  reducers: {
+    addTodo: (state, action) => {
+      state.push(action.payload);
+    },
+    completedTodo: (state, action) => {
+      const todo = state.find((todo) => todo.id === action.payload.id);
+      if (todo) {
+        todo.isCompleted = !action.payload.isCompleted;
```

Comment:
Here you can completely remove `isCompleted` from the payload since you have access to `isCompleted` property of the todo.
ex: todo.isCompleted = !todo.isCompleted